odl/diagnostics/operator.py

```diff
@@ -79,16 +82,19 @@ def norm(self):
 
         operator_norm = 0.0
         for _, x in samples(self.operator.domain):
-            result = self.operator(x)
-            x_norm = x.norm()
-            estimate = 0 if x_norm == 0 else result.norm() / x_norm
-
-            operator_norm = max(operator_norm, estimate)
+            operator_norm = max(operator_norm, self._opnorm_est(x))
 
         print('Norm is at least: {}'.format(operator_norm))
         self.operator_norm = operator_norm
         return operator_norm
 
+    def _opnorm_est(self, elem):
+        """Helper to calcualte an operator norm estimate."""
```

Comment:
Well as long as it doesnt dominate the tests its quite fine. The space has about 10 or so examples, so a number of that magnitude is fine. For the adjoint test, 10^2 evaluations are done, so anything less than 100 wouldnt be to bad, for example.